### PR TITLE
[Resolve #966] Add support for Jinja Extensions 

### LINKED ISSUE #966: Add support for Jinja Extensions
In our use of sceptre we would like to use [jinja2 extensions](https://jinja.palletsprojects.com/en/2.11.x/extensions/#).

This allows us to use custom filters and make other updates to the parser as required.

### PR DESCRIPTION
**Pull request by mrowlingfox**
Load [jinja2 extensions](https://jinja.palletsprojects.com/en/2.11.x/extensions/). From the jinja2 documentation

> Jinja supports extensions that can add extra filters, tests, globals or even extend the parser. The main motivation of extensions is to move often used code into a reusable class like adding support for internationalization.

These can be either extensions that come with jinja or custom extensions.

Example Usage:

`sceptre --j2_extension jinja2.ext.i18n --j2_extension  jinja2.ext.loopcontrols generate stack.yaml` 

## PR Checklist

- [x] Wrote a good commit message & description [see guide below].
- [x] Commit message starts with `[Resolve #issue-number]`.
- [ ] Added/Updated unit tests.
- [ ] Added/Updated integration tests (if applicable).
- [x] All unit tests (`make test`) are passing.
- [x] Used the same coding conventions as the rest of the project.
- [x] The new code passes flake8 (`make lint`) checks.
- [x] The PR relates to _only_ one subject with a clear title.
      and description in grammatically correct, complete sentences.

## Approver/Reviewer Checklist

- [x] Before merge squash related commits.

## Other Information

[Guide to writing a good commit](http://chris.beams.io/posts/git-commit/)

### CODE REVIEW

**Comment by mrowlingfox** [comment]
Any chance of getting this looked at @ngfgrant  ?

File: sceptre/cli/__init__.py [hunk 1]
```diff
@@ -45,10 +45,21 @@
     help="A YAML file of variables to template into config files.")
 @click.option(
     "--ignore-dependencies", is_flag=True, help="Ignore dependencies when executing command.")
+@click.option(
+    "--j2_extension", "j2_extensions",  multiple=True,
```
Comment (zaro0508) [inline sceptre/cli/__init__.py:49]: would it make sense to pass in a dict that can override any of the existing jinja2 environment settings including the new jinja2.extensions setting?

**Review by zaro0508 — CHANGES_REQUESTED** [review]
I think we need to add some restructured text docs for this.  Also i think we need to add one or two tests in test_cli.py to test this option,  maybe something similar to `test_ignore_dependencies_commands`,  https://github.com/Sceptre/sceptre/blob/b4d57429cf4bb7f8cf532be5cad70be5b0f37c44/tests/test_cli.py#L335

**Comment by craighurley** [comment]
Relates to #476

File: sceptre/cli/__init__.py (hunk 1 above)
Comment (mrowlingfox) [reply sceptre/cli/__init__.py:49]: Finally sat down to look into this again.I like the idea of passing all the jinj2 extensions.
@zaro0508  What are your opinions on passing in a dict via the CLI (as currently implemented) vs having it reside in the stack/stackgroup config?

**Pull request closed without merging**